src/Analyzer/NameResolver.php

```diff
@@ -148,14 +153,16 @@ public function enterNode(Node $node)
                 break;
             }
 
-            if (!($node->type instanceof FullyQualified)) {
-                foreach ($phpDocNode->getTags() as $tagValue) {
-                    if ('@' === $tagValue->name[0] && false === strpos($tagValue->name, '@var')) {
-                        $customTag = str_replace('@', '', $tagValue->name);
-                        $type = $this->resolveName(new Node\Name($customTag), Use_::TYPE_NORMAL);
-                        $node->type = $type;
+            if ($this->parseDocBlockCustomTags) {
```

Comment:
Is it possible to avoid another nested level?

Reply:
done! I also updated the docs.